### ISSUE TITLE
Sidecar not working when mtls is false

### ISSUE THREAD

**Issue by aaguilartablada** [dapr/dapr#6949]
## In what area(s)?

/area runtime

## What version of Dapr?

1.12.0-rc.3

## Expected Behavior

We have LinkerD and Dapr in our platform. We decided to use LinkerD to perform MTLS, so we deployed Dapr 1.10.5 disabling mtls and it worked fine. 

Now, when deploying dapr operator 1.12.0-rc.3 with mtls disabled it does not work:

```
global:
  mtls:
    enabled: false
```

I expect DAPR sidecar containers starting correctly the same way it started when using version 1.10.5.

## Actual Behavior

DAPR sidecar containers never start. Instead, they report the next log:

```
time="2023-09-20T14:49:27.187840991Z" level=info msg="Starting Dapr Runtime -- version 1.12.0-rc.3 -- commit af7053834669b3355fc01ce64c2a372e5787b015" app_id=product-catalog-ingestion-bystore instance=product-catalog-ingestion-bystore-6dcb9bf545-74zsm scope=dapr.runtime type=log ver=1.12.0-rc.3
time="2023-09-20T14:49:27.187872692Z" level=info msg="Log level set to: info" app_id=product-catalog-ingestion-bystore instance=product-catalog-ingestion-bystore-6dcb9bf545-74zsm scope=dapr.runtime type=log ver=1.12.0-rc.3
time="2023-09-20T14:49:27.188030494Z" level=warning msg="mTLS is disabled. Skipping certificate request and tls validation" app_id=product-catalog-ingestion-bystore instance=product-catalog-ingestion-bystore-6dcb9bf545-74zsm scope=dapr.runtime.security type=log ver=1.12.0-rc.3
time="2023-09-20T14:49:27.188298197Z" level=info msg="Initializing the operator client" app_id=product-catalog-ingestion-bystore instance=product-catalog-ingestion-bystore-6dcb9bf545-74zsm scope=dapr.runtime type=log ver=1.12.0-rc.3
time="2023-09-20T14:49:42.816327862Z" level=info msg="Received signal 'terminated'; beginning shutdown" app_id=product-catalog-ingestion-bystore instance=product-catalog-ingestion-bystore-6dcb9bf545-74zsm scope=dapr.signals type=log ver=1.12.0-rc.3
time="2023-09-20T14:49:42.816458563Z" level=fatal msg="Fatal error from runtime: context canceled: connection error: desc = \"error reading server preface: EOF\"" app_id=product-catalog-ingestion-bystore instance=product-catalog-ingestion-bystore-6dcb9bf545-74zsm scope=dapr.runtime type=log ver=1.12.0-rc.3
```

As soon as I change the configuration and enable mtls it works!!! I don't understand why.

**Comment by mukundansundar** [comment]
@aaguilartablada Could you test the fix in 1.12.0-rc.4?

**Comment by aaguilartablada** [comment]
> @aaguilartablada Could you test the fix in 1.12.0-rc.4?

It's working! Thank you so much.

**Comment by aaguilartablada** [comment]
Closing this issue.
